tests/integration/usart/main.cpp

```diff
@@ -0,0 +1,276 @@
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x06));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+
+    Usart::UsartConfigurations fig;
+    fig.tx_rx_mode= Usart::TX_RX_Mode::kDisableRX_DisableTX;
+    USART.Init(fig);
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x06));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+
+    fig.tx_rx_mode= Usart::TX_RX_Mode::kDisableRX_EnableTX;
+    USART.Init(fig);
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x08));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x06));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+
+    fig.tx_rx_mode= Usart::TX_RX_Mode::kEnableRX_DisableTX;
+    USART.Init(fig);
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x10));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x06));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+    
+   }
+   {
+
+    avr::test::AVRTest TESTER("Data size test");
+
+    uint8_t val=0;
+    USART.Init();
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x18));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x06));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+
+    Usart::UsartConfigurations fig;
+    fig.dataSize=Usart::DataSize::kFiveBits;
+    USART.Init(fig);
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x18));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UBBRL, static_cast<uint8_t>(0x33));
+
+   fig.dataSize=Usart::DataSize::kSixBits;
+    USART.Init(fig);
+    TESTER.Expect_Eq(UDR,   static_cast<uint8_t>(0x00));
+    TESTER.Expect_Eq(UCSRA, static_cast<uint8_t>(0x20));
+    TESTER.Expect_Eq(UCSRB, static_cast<uint8_t>(0x18));
+    TESTER.Expect_Eq(UBBRH, static_cast<uint8_t>(0x00));
+    val=UCSRC;
+    val=UCSRC;
+    TESTER.Expect_Eq(val, static_cast<uint8_t>(0x01));
```

Comment:
I believe the expected value should be `0x01 << 1` which is 2. As data size bit is determined by second bit in the register. `Bit 2:1 – UCSZ1:0: Character Size`. This is also for remaining values